src/assist/reflexion_agent.py

```diff
@@ -101,12 +101,57 @@ def tool_list_item(tool: BaseTool) -> str:
     return f"- {tool.name}: {tool.description}"
 
 
+def _emit_status_message(content: str) -> AIMessage:
+    """Helper to create status messages that will be emitted during streaming."""
+    return AIMessage(content=content)
+
+
+class ToolCallCallback:
+    """Callback to capture tool calls and emit status messages."""
+    def __init__(self):
+        self.tool_calls = []
+    
+    def on_tool_start(self, tool_name: str):
+        self.tool_calls.append(f"Calling tool {tool_name}.")
+    
+    def get_and_clear_calls(self):
+        calls = self.tool_calls[:]
+        self.tool_calls.clear()
+        return calls
+
+
+def _detect_tool_calls_in_messages(messages: List[BaseMessage]) -> List[str]:
```

Comment:
Why is this and the above tool call handler needed? In general, can we not just define a single callback handler to handle this whole feature for better encapsulation?